posthog/api/event.py

```diff
@@ -163,3 +167,106 @@ def values(self, request: request.Request) -> response.Response:
+                        COUNT(CASE WHEN length > 0 AND length <= 3 THEN 1 ELSE NULL END) as second,\
+                        COUNT(CASE WHEN length > 3 AND length <= 10 THEN 1 ELSE NULL END) as third,\
+                        COUNT(CASE WHEN length > 10 AND length <= 30 THEN 1 ELSE NULL END) as fourth,\
+                        COUNT(CASE WHEN length > 30 AND length <= 60 THEN 1 ELSE NULL END) as fifth,\
+                        COUNT(CASE WHEN length > 60 AND length <= 180 THEN 1 ELSE NULL END) as sixth,\
+                        COUNT(CASE WHEN length > 180 AND length <= 600 THEN 1 ELSE NULL END) as seventh,\
+                        COUNT(CASE WHEN length > 600 AND length <= 1800 THEN 1 ELSE NULL END) as eighth,\
+                        COUNT(CASE WHEN length > 1800 AND length <= 3600 THEN 1 ELSE NULL END) as ninth,\
+                        COUNT(CASE WHEN length > 3600 THEN 1 ELSE NULL END) as tenth\
+                        FROM (SELECT global_session_id, EXTRACT(\'EPOCH\' FROM (MAX(timestamp) - MIN(timestamp)))\
+                            AS length FROM ({}) as count GROUP BY 1) agg'.format(query)
+
+        result = []
+        if session_type == 'avg':
+            cursor = connection.cursor()
+            cursor.execute(overall_average_length(all_sessions), sessions_sql_params)
+            calculated = cursor.fetchall()
+            result = [{'label': 'Number of Sessions', 'count': calculated[0][0]}, {'label': 'Average Duration of Session (seconds)', 'count': round(calculated[0][1], 2)}]
+        else: 
+            dist_labels = ['0 seconds (1 event)', '0-3 seconds', '3-10 seconds', '10-30 seconds', '30-60 seconds', '1-3 minutes', '3-10 minutes', '10-30 minutes', '30-60 minutes', '1+ hours']
+            cursor = connection.cursor()
+            cursor.execute(distribution(all_sessions), sessions_sql_params)
+            calculated = cursor.fetchall()
+            result = [{'label': dist_labels[index], 'count': calculated[0][index]} for index in range(len(dist_labels))]
+        
+        return result
+
+    def _get_dates_from_request(self, request: request.Request) -> Tuple[datetime.date, datetime.date]:
+        if request.GET.get('date_from'):
+            date_from = relative_date_parse(request.GET['date_from'])
+            if request.GET['date_from'] == 'all':
+                date_from = None # type: ignore
+        else:
+            date_from = datetime.date.today() - relativedelta(days=7)
+
+        if request.GET.get('date_to'):
+            date_to = relative_date_parse(request.GET['date_to'])
+        else:
+            date_to = datetime.date.today()
+
+        # UTC is what is set in setting.py
+        if date_from is not None:
+            date_from = pd.Timestamp(date_from, tz='UTC')
+        date_to = pd.Timestamp(date_to, tz='UTC')
+        return date_from, date_to
+
+    def _filter_events(self, request: request.Request) -> Q:
```

Comment:
There's now a filter_request and filter_events. We could probably add the date logic using `request_to_date` in `_filter_request`.